lib/openxml/pptx/parts/slide.rb

```diff
@@ -1,92 +1,86 @@
-require "openxml/shapes/text"
-
 module OpenXml
   module Pptx
     module Parts
       class Slide < OpenXml::Part
-        attr_reader :layout
-        attr_accessor :relationships, :id, :shapes
-        private :relationships=, :id=, :shapes=
-
-        LAYOUT_SCHEMA =
-          "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideLayout"
+        include OpenXml::RelatablePart
+        include OpenXml::HasAttributes
+        include OpenXml::ContainsProperties
+        attr_reader :images
 
-        def initialize(layout)
-          self.relationships = OpenXml::Parts::Rels.new
-          self.layout = layout
-          self.shapes = []
-        end
-
-        private = def layout=(layout)
-          @layout = layout
-          add_relationship(layout.relationship)
-        end
+        relationship_type "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slide"
+        content_type "application/vnd.openxmlformats-officedocument.presentationml.slide+xml"
+        default_path "ppt/slides/slide.xml"
 
-        def add_relationship(relationship)
-          relationships.push(relationship)
-        end
+        attribute :show_master_shapes, displays_as: :showMasterSp, expects: :boolean
+        attribute :show_master_placeholder_animations, displays_as: :showMasterPhAnim, expects: :boolean
+        attribute :show_slide, displays_as: :show, expects: :boolean
 
-        def add_to(slide_count, ancestors)
-          parent, *rest = ancestors
-          parent.add_part rest, "slides/slide#{slide_count}.xml", self
-          parent.add_part rest, "slides/_rels/slide#{slide_count}.xml.rels", relationships
+        property :common_slide_data, required: true
+        property :color_mapping_override
 
-          layout.add_to(ancestors)
+        # = TODO ================
+        # property :transition
+        # property :timing
 
-          parent.add_slide_relationship relationship(slide_count)
+        property :extension_list
 
-          parent.add_override rest, "slides/slide#{slide_count}.xml", "application/vnd.openxmlformats-officedocument.presentationml.slide+xml"
+        LAYOUT_SCHEMA =
+          "http://schemas.openxmlformats.org/officeDocument/2006/relationships/slideLayout".freeze
 
-          shapes.each do |shape|
-            shape.add_to([self, *ancestors])
-          end
+        def initialize(*_args)
+          super
+          @images = []
```

Comment:
Why does `Slide` have an array of `images`? It _looks_ like they don't get called from anywhere.... Is it just to increment the counter of them?

Reply:
It's pretty much to get the count, though since the other parts (presentation, slide master, etc.) also have publicly-accessible arrays of their children, it's also somewhat for consistency. I can change it into just an integer counter if that would be better.